kafka/example_test.go

```diff
@@ -0,0 +1,83 @@
+package kafka
+
+import (
+	"context"
+	"time"
+
+	kafkalib "github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/sirupsen/logrus"
+)
+
+func ExampleConfig_auth() {
+	_ = Config{
+		// Append the following to your configuration (Consumer or Producer)
+		AuthType:  AuthTypeSCRAM256,
+		User:      "my-user",
+		Password:  "my-secret-password",
+		CAPEMFile: "/etc/certificate.pem",
+	}
+}
+
+func ExampleConsumer() {
+	conf := Config{
+		Topic:   "example-topic",
+		Brokers: []string{"localhost:9092"},
+		Consumer: ConsumerConfig{
+			GroupID: "example-group",
+		},
+	}
+
+	log := logrus.New()
+	c, err := NewConsumer(log, conf)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer c.Close()
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	// Consider implementing a retry mechanism.
+	for {
+		// 1. Fetch the message.
+		msg, err := c.FetchMessage(ctx)
+		if err != nil {
```

Comment:
Is there a way to capture timeouts?

Reply:
There is no timeout per se. The timeout inside the `FetchMessage` is a forced timeout, set in order to check the `ctx.Done` periodically.